runtime/lua/vim/fs.lua

```diff
@@ -144,11 +144,34 @@ end
 --- The search can be narrowed to find only files or only directories by
 --- specifying {type} to be "file" or "directory", respectively.
 ---
----@param names (string|table|fun(name: string): boolean) Names of the files
+--- Examples:
+--- <pre>lua
+--- -- location of Cargo.toml from the current buffer's path
+--- local cargo = vim.fs.find('Cargo.toml', {
+---   upward = true,
+---   stop = vim.loop.os_homedir(),
+---   path = vim.fs.dirname(vim.api.nvim_buf_get_name(0)),
+--- })
+---
+--- -- list all test directories under the runtime directory
+--- local test_dirs = vim.fs.find(
+---   {'test', 'tst', 'testdir'},
+---   {limit = math.huge, type = 'directory', path = './runtime/'}
+--- )
+---
+--- -- get all files ending with .cpp or .hpp inside lib/
+--- local cpp_hpp = vim.fs.find(function(name, path)
+---   return name:match('.*%.[ch]pp$') and path:match('[/\\\\]lib$')
+--- end, {limit = math.huge, type = 'file'})
+--- </pre>
+---
+---@param names (string|table|fun(name: string, path: string): boolean) Names of the files
 ---             and directories to find.
----             Must be base names, paths and globs are not supported.
----             The function is called per file and directory within the
----             traversed directories to test if they match {names}.
+---             Must be base names, paths and globs are not supported when {names} is a string or a table.
+---             If {names} is a function, it is called for each traversed file and directory with args:
+---             - name: base name of the current item
+---             - path: full path of the current item
+---             The function should return `true` if the given file or directory is considered a match.
```

Comment:
I guess a blank line is needed here to work around the docs rendering issue (it joins this line).
```suggestion
---
---             The function should return `true` if the given file or directory is considered a match.
```

Reply:
This does not seem to work, `gen_vimdoc.py` sends everything after the new line to the general description, just below the code examples, it renders like 

```help
...
     -- get all files ending with .cpp or .hpp inside lib/
     local cpp_hpp = vim.fs.find(function(name, path)
       return name:match('.*%.[ch]pp$') and path:match('[/\\]lib$')
     end, {limit = math.huge, type = 'file'})
<

    The function should return `true` if the given file or directory is considered a match.

    Parameters: ~
      • {names}  (string|table|fun(name: string, path: string): boolean) Names
                 of the files and directories to find. Must be base names,
                 paths and globs are not supported when {names} is a string or
                 a table. If {names} is a function, it is called for each
                 traversed file and directory with args:
                 • name: base name of the current item
                 • path: full path of the current item
...
```

Reply:
ok, I'll try to fix it later. 

Don't want to block this PR meanwhile.